cirq/sim/simulator.py

```diff
@@ -206,17 +212,15 @@ def simulate(
         Args:
             program: The circuit or schedule to simulate.
             param_resolver: Parameters to run with the program.
-            qubit_order: Determines the canonical ordering of the qubits used
-                to define the order of amplitudes in the wave function.
-            initial_state: If an int, the state is set to the computational
-                basis state corresponding to this state. Otherwise  if this
-                is a np.ndarray it is the full initial state. In this case it
-                must be the correct size, be normalized (an L2 norm of 1), and
-                be safely castable to an appropriate dtype for the simulator.
+            qubit_order: Determines the canonical ordering of the qubits. This
+                is often used in specifying the initial state, i.e. the
+                ordering of the computational basis states.
+            initial_state: The initial state for the simulation. The  form of
+                this state depends on the simulation implementation.  See
```

Comment:
Could this state dependence be packaged into a generic parameter on the class? If it causes an explosion of generics everywhere, probably best not to for now.

Reply:
this is probably the worst part of this interface right now.  In particular note the qubit order, which is not actually always necessary for some simulators.   I agree we should try to find a better solution, maybe after we get new simulators in this will become more clear.